### PR TITLE
consolidate bazel configurations and build_stamp scripts

### PR DESCRIPTION
**Pull request by ayj**

### CODE REVIEW

File: tools/bazel.rc [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # This is so we understand failures better
 build --verbose_failures
 test --test_output=errors
+test --test_size_filters=-large,-enormous
 
-build --workspace_status_command=bin/get_workspace_status --action_env=ISTIO_VERSION
-
+build --workspace_status_command=bin/get_workspace_status --action_env=ISTIO_VERSION --output_filter=^gogo
```
Comment (guptasu) [inline tools/bazel.rc]: Consider adding a comment on which component needs the output_filter. i think it is Mixer

File: tools/bazel.rc [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
 build --sandbox_tmpfs_path=/dev/shm
+
+# test --features=race
```
Comment (guptasu) [inline tools/bazel.rc]: Unclear to me why do we have this comment ?

File: mixer/bin/build_stamp.sh [hunk 3]
```diff
@@ -1,45 +0,0 @@
...
-
-# Check for local changes
-git diff-index --quiet HEAD --
-if [[ $? == 0 ]];
-then
-    tree_status="Clean"
-else
-    tree_status="Modified"
-fi
-echo "buildStatus ${tree_status}"
```
Comment (guptasu) [inline mixer/bin/build_stamp.sh]: buildStatus seems to be missing from the new echo statements. Is it intentional ?

File: tools/bazel.rc (hunk 1 above)
Comment (kyessenov) [reply tools/bazel.rc]: care to educate what it does and why it's needed?

Comment (guptasu) [reply tools/bazel.rc]: Mixer has some extra paths mentioned in the gogo proto library build rule. For example `../../api`, `../api`, `api` in order for protos living outside Mixer repo to compile correctly. If the path is not used, gogo issues warning and this suppression is for not showing those warning about `path ../../api not used...`. I think we have a TODO or may be a bug to get rid of this kind of multiple paths listing.

Comment (kyessenov) [reply tools/bazel.rc]: thanks! yeah, I know this warning.

**Comment by mattdelco** [comment]
Looks like my PR https://github.com/istio/istio/pull/1902 overlaps with this one but each lacks changes/fixes that's in the other.

**Comment by ayj** [comment]
@mattdelco, I'm happy to drop this PR if you want to roll in any missing changes into yours.

**Comment by mattdelco** [comment]
I'll work on trying to make mine a superset.  I think the main thing I have to add is having a few more things sent to stdout by get_workspace_status.

**Comment by ayj** [comment]
Superseded by https://github.com/istio/istio/pull/1902

**Pull request closed without merging**